packages/block-editor/src/components/tool-selector/index.js

```diff
@@ -31,11 +31,28 @@ const selectIcon = (
 );
 
 function ToolSelector( props, ref ) {
-	const mode = useSelect(
-		( select ) => select( blockEditorStore ).__unstableGetEditorMode(),
-		[]
-	);
-	const { __unstableSetEditorMode } = useDispatch( blockEditorStore );
+	const [ isSimpleMode, setIsSimpleMode ] = useState( false );
+	const [ originalTemplateLocks, setOriginalTemplateLocks ] = useState( {} );
+	const { mode, blocksWithinMainBlockClientIds, getBlockAttributes } =
+		useSelect( ( select ) => {
+			const {
+				__unstableGetEditorMode,
+				getBlockOrder,
+				getBlockAttributes: _getBlockAttributes,
+			} = select( blockEditorStore );
+
+			const mainBlockClientId = getBlockOrder()[ 1 ];
+
+			return {
+				mode: __unstableGetEditorMode(),
+				blocksWithinMainBlockClientIds:
+					getBlockOrder( mainBlockClientId ),
```

Comment:
We could use [this technique](https://github.com/WordPress/gutenberg/pull/64628#pullrequestreview-2265729298) as well.